frontend/src/components/MainNavigation.vue

```diff
@@ -292,6 +330,57 @@ export default {
         name = fallback
       }
       return !this.namespaced ? { name, query: { namespace } } : { name, params: { namespace } }
+    },
+    onInputProjectFilter () {
+      this.selectedProjectIndex = 0
+      this.numberOfVisibleProjects = 10
+      if (this.projectFilterHasExactMatch) {
+        this.selectedProjectIndex = findIndex(this.sortedAndFilteredProjectListWithAllProjects, { metadata: { name: this.projectFilter } })
+      } else if (this.sortedAndFilteredProjectList.length === 1) {
+        const project = head(this.sortedAndFilteredProjectList)
+        this.selectedProjectIndex = findIndex(this.sortedAndFilteredProjectListWithAllProjects, project)
+      }
+      this.scrollSelectedProjectIntoView()
+    },
+    selectProjectWithKeys (keyDirection) {
+      if (keyDirection === 'up') {
+        if (this.selectedProjectIndex > 0) {
+          this.selectedProjectIndex--
+        }
+      } else if (keyDirection === 'down') {
+        if (this.selectedProjectIndex < this.sortedAndFilteredProjectListWithAllProjects.length - 1) {
+          this.selectedProjectIndex++
+        }
+      }
+
+      if (this.selectedProjectIndex >= this.numberOfVisibleProjects - 1) {
+        this.numberOfVisibleProjects++
+      }
+
+      this.scrollSelectedProjectIntoView()
+    },
+    scrollSelectedProjectIntoView () {
+      const projectIndexInVisibleList = findIndex(this.visibleProjectList, this.selectedProject)
+      const el = get(this.$refs.projectList.$children[projectIndexInVisibleList], '$el')
+
+      if (el) {
+        el.scrollIntoView(false)
+      }
+    },
+    handleProjectListScroll (event) {
+      const parentEl = this.$refs.projectList.$el
+      const parentYPos = parentEl.getBoundingClientRect().top + parentEl.getBoundingClientRect().height
+      const lastProjectEl = get(last(this.$refs.projectList.$children), '$el')
```

Comment:
lastProjectEl could be undefined, right? -> it would brake in the next line

Reply:
yes- could be. If the project list is empty. Let's review this method in detail

Reply:
- add checks for undefined / not set
- el -> element

Reply:
done